bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -28,215 +28,203 @@ spec:
-        # ovn db is stored in the pod in /etc/openvswitch (default ovs tool location)
-        # and on the host in /var/lib/openvswitch/
         - mountPath: /etc/openvswitch/
-          name: host-var-lib-ovs
+          name: host-var-lib-ovn
         - mountPath: /var/lib/openvswitch/
-          name: host-var-lib-ovs
-        - mountPath: /var/run/openvswitch/
-          name: host-var-run-ovs
-        - mountPath: /var/run/kubernetes/
-          name: host-var-run-kubernetes
-
+          name: host-var-lib-ovn
+        - mountPath: /run/openvswitch/
+          name: host-run-ovn
+        - mountPath: /env
+          name: env-overrides
         resources:
           requests:
             cpu: 100m
             memory: 300Mi
+      - name: sbdb
+        image: {{.OvnImage}}
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          if [[ -f "/env/${K8S_NODE}" ]]; then
+            set -o allexport
+            source "/env/${K8S_NODE}"
+            set +o allexport
+          fi
+          exec /usr/share/openvswitch/scripts/ovn-ctl run_sb_ovsdb \
+            --no-monitor \
+            --ovn-nb-log="-vconsole:${OVN_LOG_LEVEL} -vfile:off"
         env:
-        - name: OVN_DAEMONSET_VERSION
-          value: "3"
-        - name: OVN_LOG_SB
-          value: "-vconsole:info -vfile:info"
-        - name: OVN_NB_PORT
-          value: "9641"
-        - name: OVN_SB_PORT
-          value: "9642"
-        - name: OVN_NET_CIDR
-          valueFrom:
-            configMapKeyRef:
-              name: ovn-config
-              key: net_cidr
-        - name: OVN_SVC_CIDR
-          valueFrom:
-            configMapKeyRef:
-              name: ovn-config
-              key: svc_cidr
         - name: K8S_NODE
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
-        - name: K8S_APISERVER
-          valueFrom:
-            configMapKeyRef:
-              name: ovn-config
-              key: k8s_apiserver
-        - name: OVN_KUBERNETES_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace
-      # end of container
-
+        - name: OVS_LOG_LEVEL
+          value: info
+        volumeMounts:
+        - mountPath: /etc/openvswitch/
+          name: host-var-lib-ovn
+        - mountPath: /var/lib/openvswitch/
+          name: host-var-lib-ovn
+        - mountPath: /run/openvswitch/
+          name: host-run-ovn
+        - mountPath: /env
+          name: env-overrides
       - name: ovnkube-master
         image: {{.OvnImage}}
-
-        command: ["/root/ovnkube.sh", "ovn-master"]
-
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          if [[ -f "/env/${K8S_NODE}" ]]; then
+            set -o allexport
+            source "/env/${K8S_NODE}"
+            set +o allexport
+          fi
+          retries=0
+          while true; do
+            if [[ ! -f /var/run/openvswitch/ovn-northd.pid ]]; then
```

Comment:
I don't think this is reliable; there may be a stale ovn-northd.pid file

Reply:
There's no easy way to tell if ovn-northd is running. We no longer run the host pid namespace, so we can't watch the pid.

I'm also not sure we need to wait for northd.

Reply:
`ovnkube.sh` does

    ovs-appctl -t /var/run/openvswitch/ovn-northd.${northd_pid}.ctl version &> /dev/null

Reply:
I don't see any northd dependency in ovnkube. More precisely, the only code that talks to sbdb is the node process.

Reply:
Removing the ready check and everything comes up fine, even with a decent amount of chaos-monkeying